File: src/main/java/com/portfolio/main/presentation/rest/account/user/response/UserResponse.java
package com.portfolio.main.presentation.rest.account.user.response;

import com.portfolio.main.application.login.dto.UserDto;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
public class UserResponse {
    private Long id;
    private String loginId;
    private String userName;
    private LocalDateTime createdAt;
    private List<String> roleCodes;

    public UserResponse(UserDto userDto) {
        this.id = userDto.getId();
        this.loginId = userDto.getLoginId();
        this.userName = userDto.getUserName();
        this.createdAt = userDto.getCreatedAt();
        this.roleCodes = userDto.getRoleCodes();
    }

}

File: src/main/resources/static/js/apis/response/user/UserResponse.js
const tag = `[UserResponse]`;

export default class UserResponse {
    #id;
    #loginId;
    #userName;
    #roleCodes;

    constructor({id, loginId, userName, roleCodes}) {
        this.#id = id;
        this.#loginId = loginId;
        this.#userName = userName;
        this.#roleCodes = roleCodes;

        console.log(`${tag} :: `)
        console.dir(this);
    }

    getId() {
        return this.#id;
    }

    getLoginId() {
        return this.#loginId;
    }

    getUserName() {
        return this.#userName;
    }

    getRoleCodes() {
        return this.#roleCodes;
    }
}
